lib/action_view/component/render_monkey_patch.rb

```diff
@@ -15,15 +15,27 @@ def render(options = {}, args = {}, &block)
 
           options.render_in(self, &block)
         elsif options.is_a?(Class) && options < ActionView::Component::Base
-          options.new(args).render_in(self, &block)
+          safe_new_component(options, args).render_in(self, &block)
         elsif options.is_a?(Hash) && options.has_key?(:component)
-          options[:component].new(options[:locals]).render_in(self, &block)
+          safe_new_component(options[:component], options[:locals]).render_in(self, &block)
         elsif options.respond_to?(:to_component_class) && !options.to_component_class.nil?
           options.to_component_class.new(options).render_in(self, &block)
```

Comment:
Do we need to update this case as well?

Reply:
I went [back and forth on this](https://github.com/github/actionview-component/pull/171#issuecomment-570830559), and I'm leaning towards excluding this case, even though it's sort of weird. But I'd love to hear your thoughts.

If we use `#safe_new_component` for the `#to_component_class` case:
* `options` will be  the object on which `#to_component_class` is called. Which I think means we _always_ want the non-zero arity initializer to be called.
* `#safe_new_component` calls `#blank?` on `options`. And if `options` is a model or whatever responds to `:to_component_class`, I'm not sure if calling `#blank?` is right.
* If we drop the `#blank?` check in `#safe_new_component` all-together and only check the arity of `#initialize` then we run the risk of accidentally dropping arguments a user passed to initialize, and hiding a helpful `ArgumentError`.

If we don't use `#safe_new_component`
* The call to `.new`, is now made in two separate methods (`#render` and `#safe_new_component`). Which feels less ideal.

In general this change I've made feels a little hack-y. But I'm not sure what sort of change would be better.  Do you have any ideas?

It still seems important to me that one can either we support 0-arity initializers or give a more leading error message. Since users aren't calling `#new` on their component, it's not immediately clear why `render(MyComponent)` raises `"Argument Error: wrong number of arguments (1 for 0)"`

Reply:
❤️ thanks for the explanation. I think you're right to take the approach you did here. 

Mind dropping `options will be the object on which #to_component_class is called. Which I think means we always want the non-zero arity initializer to be called.` in as a code comment above this line?

Reply:
✅ Done. Good thinking! Let me know if you think the comment could be clearer. Right now I have:

> In this case, the `options` passed to `new` is the object on which `#to_component_class` is called. So we want the non-zero arity initializer to be called.